src/main/webapp/app/propertyFilter.js

```diff
@@ -0,0 +1,31 @@
+angular
+    .module('app')
+    .filter('propertyFilter', function() { return PropertyFilter});
+
+function PropertyFilter(items, props) {
```

Comment:
It's just magic function from stackoveflow. We need this for autocomplete.
When you are typing in select box text field, we should hide such items, which don't suit us, just filter them.
By default we can filter items this way `item | filter { name: $select.search, age: $select.search}`. It works like **and** for name and age. Using default filter we say to ui-select: hey, just show me all items where age fits entered string and **at the same time** age fits entered string.
But we need **or** here. So with this strange function we can ask again: hey, show me such items where name fits entered string **or** age fits one. 

P.S. Nice comment. Finally realized why this function is needed :)